cmd/cloud.go

```diff
@@ -360,6 +371,52 @@ Examples:
 			deploymentId = "/" + deploymentId
 		}
 
+		// get a tag, ask the user for a tag if they didn't provide one or use the commit hash if we're in CI
+		tag, _ := cmd.Flags().GetString("tag")
+		if tag == "" {
+			if tui.HasTTY {
+				randomTag := util.RandStringBytes(8)
+				tag = tui.Select(logger, "Select a tag", "", []tui.Option{
+					tui.Option{
+						Text: "latest",
+						ID:   "latest",
+					}, {
+						Text: randomTag,
+						ID:   randomTag,
+					}, {
+						Text: "Enter manually",
+						ID:   "__custom",
+					}})
+
+				if tag == "__custom" {
+					tag = tui.Input(logger, "Enter a custom tag", "")
+					if tag == "" {
+						errsystem.New(errsystem.ErrDeployProject, fmt.Errorf("tag cannot be empty"),
+							errsystem.WithContextMessage("Tag cannot be empty")).ShowErrorAndExit()
+					}
+					tag = slug.Make(tag)
+				}
+			} else {
+				if ciCommit != "" {
+					tag = ciCommit
+				} else {
+					tag = "latest"
+				}
+			}
+		}
+
+		// get a tag message, ask the user for a tag message if they didn't provide one or use the commit message if we're in CI
+		tagMessage, _ := cmd.Flags().GetString("tag-message")
+		if tagMessage == "" {
```

Comment:
i don't think we want to force a tag message from a required input. i think we set it to empty string.

same with the above. imply "latest" when you go to deploy if not provided from command line.